src/test/scala/chiselTests/PrefixSpec.scala

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.experimental.hierarchy.{instantiable, public, Definition, Instance, Instantiate}
+import circt.stage.ChiselStage.emitCHIRRTL
+
+class PrefixSpec extends ChiselFlatSpec with ChiselRunners with Utils with MatchesAndOmits {
+
+  behavior.of("withModulePrefix")
+
+  it should "prefix modules in a withModulePrefix block, but not outside" in {
+    class Foo extends RawModule {
+      val a = Wire(Bool())
+    }
+
+    class Top extends RawModule {
+      val foo = Module(new Foo)
+
+      withModulePrefix("Pref") {
+        val pref_foo = Module(new Foo)
+      }
+    }
+
```

Comment:
can you please show a test that does two nested prefix blocks (make sure it's Foo_Bar not Bar_Foo and so on!)

Reply:
This is what `it should "Allow nested module prefixes"` does.